src/nidas/dynld/raf/UDPiPMSensor.cc

```diff
@@ -0,0 +1,218 @@
+// -*- mode: C++; indent-tabs-mode: nil; c-basic-offset: 4; tab-width: 4; -*-
+// vim: set shiftwidth=4 softtabstop=4 expandtab:
+/*
+ ********************************************************************
+ ** NIDAS: NCAR In-situ Data Acquistion Software
+ **
+ ** 2008, Copyright University Corporation for Atmospheric Research
+ **
+ ** This program is free software; you can redistribute it and/or modify
+ ** it under the terms of the GNU General Public License as published by
+ ** the Free Software Foundation; either version 2 of the License, or
+ ** (at your option) any later version.
+ **
+ ** This program is distributed in the hope that it will be useful,
+ ** but WITHOUT ANY WARRANTY; without even the implied warranty of
+ ** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ ** GNU General Public License for more details.
+ **
+ ** The LICENSE.txt file accompanying this software contains
+ ** a copy of the GNU General Public License. If it is not found,
+ ** write to the Free Software Foundation, Inc.,
+ ** 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ **
+ ********************************************************************
+*/
+
+#include "UDPiPMSensor.h"
+
+#include <nidas/util/UTime.h>
+#include <nidas/util/Logger.h>
+
+#include <csignal>
+#include <unistd.h>
+#include <sys/wait.h>
+
+using namespace std;
+using namespace nidas::dynld::raf;
+
+namespace n_u = nidas::util;
+
+NIDAS_CREATOR_FUNCTION_NS(raf, UDPiPMSensor);
+
+const int UDPiPMSensor::MAX_CHANNELS = 8;
+
+
+UDPiPMSensor::UDPiPMSensor() :
+    _deviceAddr(), _statusPort(0), _measureRate(0), _recordPeriod(0),
+    _baudRate(0), _numAddr(0), _addrInfo(8), _ctrl_pid(0)
+{
+
+}
+
+UDPiPMSensor::~UDPiPMSensor()
+{
+    close();
+}
+
+void UDPiPMSensor::validate()
+{
+    UDPSocketSensor::validate();
+
+    const Parameter *p;
+    p = getParameter("device"); // device
+    if (!p) throw n_u::InvalidParameterException(getName(),
+          "device", "not found");
+    _deviceAddr = p->getStringValue(0);
+    ILOG(("device is %s", _deviceAddr.c_str()));
```

Comment:
FWIW, nidas logging supports stream output operators, but it's not used as often because there is so much code using the older percent formats.  For example, this code can be written like so:
```
ILOG(("device is ") << _deviceAddr);
```